docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+
+    type Bucket struct {
+        ID            string `json:"id,omitempty" codec:"id,omitempty"`
+        Created       int64  `json:"created,omitempty" codec:"created,omitempty"` 
+        Modified      int64  `json:"modified,omitempty" codec:"modified,omitempty"`
+        Service       string `json:"device,omitempty" codec:"device,omitempty"`
+        Name          string `json:"name,omitempty" codec:"name,omitempty"`  // metric or CPE name key
+        Value         string `json:"value,omitempty" codec:"value,omitempty"` // metric or CPE value
+        ValueType     string `json:"valueType,omitempty" codec:"valueType,omitempty"`
+        Description   string `json:"name,omitempty" codec:"description,omitempty"`  // human readable details
+    }
+  ```
+
+- Proposed endpoint for the REST endpoints that respond with what metrics and CPE the service offers are:  /api/v2/metrics and /api/v2/cpe
+- Configuration, not unlike that provided in core data, will specify what message endpoints the metrics and CPE messages should be sent.
+  - Proposed configuration for each service for metric and CPE endpoints:
+
+  ``` yaml
+    [MetricsMessageQueue]
+    Protocol = 'tcp'
+    Host = '*'
+    Port = 5573
+    Type = 'zero'
+    Topic = 'metrics'
+    [MessageQueue.Optional]
+        # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+        # Client Identifiers
+        Username =""
+        Password =""
+        ClientId ="service-abc"
+        # Connection information
+        Qos = "0" # Quality of service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive =  "10" # Seconds (must be 2 or greater)
+        Retained = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+        SkipCertVerify = "false"
+
+    [CPEMessageQueue]
+    Protocol = 'tcp'
+    Host = '*'
+    Port = 5574
+    Type = 'zero'
+    Topic = 'cpe'
+    [MessageQueue.Optional]
+        # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+        # Client Identifiers
+        Username =""
+        Password =""
+        ClientId ="service-abc"
+        # Connection information
+        Qos = "0" # Quality of service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive =  "10" # Seconds (must be 2 or greater)
+        Retained = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+        SkipCertVerify = "false"
+  ```
+
+- Metrics and CPE collected/monitored and reported by a service will be associated with a name key.  That name key will be used in the configuration to turn `on` the metric or CPE and will be used in the model/DTO when reporting on the metric or CPE. 

```

Comment:
does this mean each individual metric and CPE can be turned `on` or `off`?

Reply:
this can be accomplished with putting the  service key in the topic like `edgex/metrics/edgex-core-data` . This allows subscriber to filter for a specific service using the full topic or all metrics using a wild card  `edgex/metrics#`

Either way the subscriber knows receives the full topic each time and can extract the service key from it.

Reply:
Add this to the ADR.  Open to suggestions on best approach and as implementation detail.